solutions/fully-configurable/variables.tf

```diff
@@ -0,0 +1,492 @@
+    condition = length(distinct(
+      flatten([
+        # Check through rules
+        for rule in flatten([var.network_acls[*].rules]) :
+        # Return false if direction is not valid
+        false if !can(regex("^([a-z]|[a-z][-a-z0-9]*[a-z0-9])$", rule.name))
+      ])
+    )) == 0
+  }
+
+}
+
+variable "clean_default_sg_acl" {
+  description = "Remove all rules from the default VPC security group and VPC ACL (less permissive)"
+  type        = bool
+  default     = false
+}
+
+variable "address_prefixes" {
+  description = "The IP range that will be defined for the VPC for a certain location. Use only with manual address prefixes"
+  type = object({
+    zone-1 = optional(list(string))
+    zone-2 = optional(list(string))
+    zone-3 = optional(list(string))
+  })
+  default = {
+    zone-1 = null
+    zone-2 = null
+    zone-3 = null
+  }
+  validation {
+    error_message = "Keys for `use_public_gateways` must be in the order `zone-1`, `zone-2`, `zone-3`."
+    condition = var.address_prefixes == null ? true : (
+      (length(var.address_prefixes) == 1 && keys(var.address_prefixes)[0] == "zone-1") ||
+      (length(var.address_prefixes) == 2 && keys(var.address_prefixes)[0] == "zone-1" && keys(var.address_prefixes)[1] == "zone-2") ||
+      (length(var.address_prefixes) == 3 && keys(var.address_prefixes)[0] == "zone-1" && keys(var.address_prefixes)[1] == "zone-2") && keys(var.address_prefixes)[2] == "zone-3"
+    )
+  }
+}
+
+##############################################################################
+# Add routes to VPC
+##############################################################################
+
+variable "routes" {
+  description = "Allows you to specify the next hop for packets based on their destination address"
+  type = list(
+    object({
+      name                          = string
+      route_direct_link_ingress     = optional(bool)
+      route_transit_gateway_ingress = optional(bool)
+      route_vpc_zone_ingress        = optional(bool)
+      routes = optional(
+        list(
+          object({
+            action      = optional(string)
+            zone        = number
+            destination = string
+            next_hop    = string
+          })
+      ))
+    })
+  )
+  default = []
+}
+
+##############################################################################
+# VPC Flow Logs
+##############################################################################
+
+variable "enable_vpc_flow_logs" {
+  description = "To enable vpc flow logs, set this to true."
+  type        = bool
+  default     = false
+}
+
+variable "skip_vpc_cos_authorization_policy" {
+  description = "To skip creating an IAM authorization policy that allows the VPC to access the Cloud Object Storage, set this variable to `true`."
+  type        = bool
+  default     = true
+}
+
+variable "existing_cos_instance_crn" {
+  description = "CRN of the existing COS instance. It is required to create the bucket used for flow logs."
```

Comment:
You should mention it is only required if `enable_vpc_flow_logs` is set to true in the description

Reply:
added